chainermn/util.py

```diff
@@ -0,0 +1,59 @@
+import os
+import sys
+
+
+_orig_exc_hook = None
+
+
+def _global_except_hook(exctype, value, traceback):
+    """Catches an unhandled exception and call MPI_Abort()."""
+    try:
+        if _orig_exc_hook:
+            _orig_exc_hook(exctype, value, traceback)
+        else:
+            sys.__excepthook__(exctype, value, traceback)
+
+    finally:
+        import mpi4py.MPI
+        rank = mpi4py.MPI.COMM_WORLD.Get_rank()
+        sys.stderr.write("\n")
+        sys.stderr.write("******************************************\n")
+        sys.stderr.write("ChainerMN: \n")
+        sys.stderr.write("   Uncaught exception on rank {}. \n".format(rank))
+        sys.stderr.write("   Calling MPI_Abort() to shut down MPI...\n")
+        sys.stderr.write("******************************************\n")
+        sys.stderr.write("\n\n")
+        sys.stderr.flush()
+
+        try:
+            import mpi4py.MPI
+            mpi4py.MPI.COMM_WORLD.Abort(1)
+        except Exception as e:
+            # Something is completely broken...
+            # There's nothing we can do any more ¯\_(ツ)_/¯
+            sys.stderr.write(
+                "Sorry, failed to stop MPI and the process may hang.\n")
+            sys.stderr.flush()
+            raise e
+
+
+#
```

Comment:
Code comments below look duplicate with additional faq? More specific comment would be nice as docstring for this function added to docs in `docs/source/reference/index.rst` .

Reply:
I added a `configuration` section to the reference manual, although there are only 1 entry of `CHAINERMN_FORCE_ABORT_ON_EXCEPTION` .